### ISSUE TITLE
Similar evaluation results of different loss metrics in PASCAL VOC dataset

### ISSUE THREAD

**Issue by niranjanfall2017** [ultralytics/yolov5#7047]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi,
    I wanted to observe the performance of the YOLOv5 model on different loss metrics such as IOU, GIOU, DIOU, and CIOU.
I trained the YOLOv5 model on the PASCAL VOC with each of the loss metric individually 
(By choosing different metrics in this line iou = bbox_iou(pbox.T, tbox[i], x1y1x2y2=False, CIoU=True)  # iou(prediction, target)   of computeloss function). 

1. Hyperparameters were kept the same
2. 150 epochs with batch size 16 

When I try to evaluate the trained models(In my case its 4, one for each metric), I do not see much difference in evaluation results. I am able to see very similar performance in IoU as compared to DIoU.  

IOU NMS was used as the evaluation metric in all cases. 

It would be helpful to know if anyone else has observed the same? But in the previous versions of yolo, there exists a significant difference in evaluation results for the same PASCAL VOC dataset.

**Comment by vvision0** [comment]
I have the same question, the difference of Map is usually 0.001 and sometimes the IoU loss may have better performence than CIoU, but what i'm concerned is that my map and map0.5:0.95 are 0.824 and 0.579 on windows, 0.832 on COLAB, without any 
model change.The official result is about 0.86, I don't know why my result is lower.

**Comment by niranjanfall2017** [comment]
Yes, the issue exists. I did some experiments and understood if we train a model for more epochs (say 150 ) irrespective of the metric, the model converges and gives the same results. (Strange but I don't couldn't find a reason)

But if you train for less epochs say 50 or so, you would be able to see some difference. I was able to observe this. 

Hyperparameters choice is the reason for lower results than official repo. Most of the repo don't mention all or in yolov5 we might be adding addition ones which causes the difference. 

Again, this is from my understanding and observations. Could be wrong as well.

**Comment by glenn-jocher** [comment]
@niranjanfall2017 all of the IoU regression losses are naturally closely related and I wouldn't expect major differences in results between them. In most cases on COCO they may produce differences far less than 0.01 mAP.

@vvision0 VOC commands to reproduce are clearly indicated in Colab [notebook](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb#scrollTo=BSgFCAcMbk1R&line=2&uniqifier=1) Appendix section: 

```python
# VOC
for b, m in zip([64, 64, 32, 16], ['yolov5s', 'yolov5m', 'yolov5l', 'yolov5x']):  # zip(batch_size, model)
  !python train.py --batch {b} --weights {m}.pt --data VOC.yaml --epochs 50 --cache --img 512 --hyp hyp.VOC.yaml --project VOC --name {m}
```

**Comment by niranjanfall2017** [comment]
@glenn-jocher  Thank you for your inputs, its cleared my confusion on the performance.
